http3/client.go

```diff
@@ -196,9 +196,12 @@ func (c *ClientConn) roundTrip(req *http.Request) (*http.Response, error) {
 		// wait for the handshake to complete
 		earlyConn, ok := c.Connection.(quic.EarlyConnection)
 		if ok {
+			trace := httptrace.ContextClientTrace(req.Context())
 			select {
 			case <-earlyConn.HandshakeComplete():
+				traceTLSHandshakeDone(trace, earlyConn.ConnectionState().TLS, nil)
 			case <-req.Context().Done():
+				traceTLSHandshakeDone(trace, earlyConn.ConnectionState().TLS, req.Context().Err())
```

Comment:
This is not correct. A context cancellation is not equivalent to a TLS handshake failure.